crates/iota-indexer/src/backfill/mod.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{ops::RangeInclusive, sync::Arc};
+
+use async_trait::async_trait;
+use clap::{Subcommand, ValueEnum};
+
+use crate::{backfill::sql::sql_backfill::SqlBackfill, db::ConnectionPool, errors::IndexerError};
+
+pub(crate) mod ingestion;
+pub mod runner;
+pub(crate) mod sql;
+
+/// Encapsulates the logic to process and persist data for a chunk of
+/// checkpoints.
+#[async_trait]
```

Comment:
Do we need to use the `async_trait` now that `async fn` is stable in traits?

Reply:
Fixed in https://github.com/iotaledger/iota/pull/7603

Reply:
Only if dynamic dispatch (`dyn Trait`) is required. Native async trait methods, despite being stable, are not yet directly compatible with `dyn Trait` without explicit future boxing. We could remove `async_trait` by manually adding minimal boilerplate. However, since `async_trait` is still used in this crate and explicitly solves this issue ([docs.rs](https://docs.rs/async-trait/latest/async_trait/)), I suggest we can reconsider this dependency optimization at a later stage.